### PR TITLE
MINOR: Add test for verifying retention on changelog topics

### PR DESCRIPTION
**Pull request by cadonna**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/WindowedChangelogRetentionIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+    private KafkaStreams kafkaStreams;
+    private String streamOneInput;
+    private String streamTwoInput;
+    private String outputTopic;
+    private KGroupedStream<String, String> groupedStream;
+
+    @Rule
+    public TestName testName = new TestName();
+
+    @Before
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/integration/WindowedChangelogRetentionIntegrationTest.java:94]: Could we define it as `@BeforeClass`? I think we do not need to create input/output topics for each test case.

**Review by guozhangwang — APPROVED** [review]
The added integration test LGTM. Just have one minor comment to reduce the test time.

File: streams/src/test/java/org/apache/kafka/streams/integration/WindowedChangelogRetentionIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+            .toStream()
+            .to(outputTopic, Produced.with(windowedSerde, Serdes.Long()));
+
+        startStreams();
+
+        verifyChangelogRetentionOfWindowedStore(storeName, expectedRetention);
+    }
+
+    @Test
+    public void joinWindowedChangelogShouldHaveRetentionOfDoubleWindowSizeIfWindowSizeLargerThanDefaultRetention() throws Exception {
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/integration/WindowedChangelogRetentionIntegrationTest.java:227]: Should we verify for the case if user specify `before/after` explicitly, and differently?

Comment (cadonna) [reply streams/src/test/java/org/apache/kafka/streams/integration/WindowedChangelogRetentionIntegrationTest.java:227]: Good idea!

File: streams/src/test/java/org/apache/kafka/streams/integration/WindowedChangelogRetentionIntegrationTest.java (hunk 1 above)
Comment (cadonna) [reply streams/src/test/java/org/apache/kafka/streams/integration/WindowedChangelogRetentionIntegrationTest.java:94]: Good point!

**Comment by guozhangwang** [comment]
LGTM. @cadonna please feel free to merge after green builds.

**Comment by cadonna** [comment]
Test failures are unrelated.

**Pull request merged**